Lib/email/policy.py

```diff
@@ -140,6 +146,7 @@ def header_store_parse(self, name, value):
         """
         if hasattr(value, 'name') and value.name.lower() == name.lower():
             return (name, value)
+        validate_header_name(name)
```

Comment:
Should we validate the header name if the value has a "name"?

Reply:
Do you mean it would better to change the function name to `validate_header(name)` ?

Reply:
No, I meant: should we also call `validate_header_name` in the `if hasattr(value, 'name') and value.name.lower() == name.lower():` before returning `(name, value)` as is?